plugin/src/Reconciler/init.lua

```diff
@@ -64,8 +71,29 @@ function Reconciler:applyPatch(patch)
 		end
 	end
 
+	local patchTimestamp = DateTime.now():FormatLocalTime("LTS", "en-us")
+
 	local unappliedPatch = applyPatch(self.__instanceMap, patch)
 
+	if self.__fetchOnPatchFail then
+		-- TODO Is it worth doing this for additions that fail?
+		-- It seems unlikely that a valid Instance can't be made with `Instance.new`
+		-- but can be made using GetObjects
+		if PatchSet.hasUpdates(unappliedPatch) then
+			local idList = table.create(#unappliedPatch.updated)
+			for i, entry in unappliedPatch.updated do
+				idList[i] = entry.id
+			end
+			-- TODO this is destructive to any properties that are
+			-- overwritten by the user but not known to Rojo. We can probably
+			-- mitigate that by keeping tabs of any instances we need to swap.
+			fetchInstances(idList, self.__instanceMap, self.__apiContext)
```

Comment:
I don't know if it makes sense to be performing HTTP requests in the reconciler. I suppose the entire fetch process could conceptually be seen as part of patch application, but I think I'd rather have the API call happen in ServeSession (maybe by wrapping Reconciler:applyPatch in a new function ServeSession:__applyPatch) to better follow the existing architecture, to make the control flow a little clearer, and to avoid passing an ApiContext plus a setting to the reconciler

Reply:
I don't love it, but I'll just be honest: this was exploratory surgery on the plugin. I didn't really know where to put it, so I went with the reconciler because it was at least sensible.